src/fund/policies/risk-management/MaxConcentration.sol

```diff
@@ -20,19 +21,46 @@ contract MaxConcentration is TradingSignatures, DSMath {
         maxConcentration = _maxConcentration;
     }
 
+    function _calcConcentration(uint assetGav, uint totalGav) internal returns (uint) {
+        return mul(assetGav, ONE_HUNDRED_PERCENT) / totalGav;
+    }
+
     function rule(bytes4 sig, address[5] calldata addresses, uint[3] calldata values, bytes32 identifier)
         external
         returns (bool)
     {
         Accounting accounting = Accounting(Hub(Trading(msg.sender).hub()).accounting());
         address denominationAsset = accounting.DENOMINATION_ASSET();
         // Max concentration is only checked for non-quote assets
-        address incomingToken = (sig == TAKE_ORDER) ? addresses[2] : addresses[3];
-        if (denominationAsset == incomingToken) { return true; }
-        uint concentration = mul(
-            accounting.calcAssetGAV(incomingToken),
-            ONE_HUNDRED_PERCENT
-        ) / accounting.calcGav();
+        address takerToken = (sig == TAKE_ORDER) ? addresses[2] : addresses[3];
+        if (denominationAsset == takerToken) { return true; }
+
+        uint concentration;
+        uint totalGav = accounting.calcGav();
+        if (sig == MAKE_ORDER) {
+            IPriceSource priceSource = IPriceSource(Hub(Trading(msg.sender).hub()).priceSource());
+            address makerToken = addresses[2];
+            uint makerQuantiyBeingTraded = values[0];
+            address takerToken = addresses[3];
```

Comment:
somewhat duplicate of line 35; can remove line 35 and put in `else` block here as well